### PR TITLE
doc: matter: remove OTBR guides from Matter docset

### PR DESCRIPTION
**Pull request by greg-fer**
Removed deprecated guide from the Matter docset.
KRKNWK-15836.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/matter/index.rst [hunk 1]
```diff
@@ -16,6 +41,6 @@ To read about Matter in the nRF Connect SDK, see `Matter protocol <https://devel
    BUILDING
    nrfconnect_*
    android_*
-   openthread_*
+   openthread_rcp_nrf_dongle
```
Comment (LuDuda) [inline doc/matter/index.rst:44]: Let's remove this one as well. The instruction exists in NCS's Thread Border Router one.

**Review by LuDuda — COMMENTED** [review]
Thanks! One nit before approving.

**Review by LuDuda — APPROVED** [review]

**Comment by greg-fer** [comment]
@LuDuda , I've removed other pages too, related to Android. Can you double-check?

**Review by LuDuda — APPROVED** [review]
Perfect!

**Review by gmarull — APPROVED** [review]

**Pull request merged**